pkg/kv/kvserver/intentresolver/intent_resolver_integration_test.go

```diff
@@ -404,6 +406,8 @@ func TestIntentResolutionUnavailableRange(t *testing.T) {
 			TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
 				// Configure so that intent resolution on t2 cannot finish
 				// successfully and blocks, relying on the timeout to unblock.
+				t2RangeIDMu.Lock()
```

Comment:
Do we want to hold this mutex until the request is cancelled? Could this lead to a test deadlock? Should we instead make access to `t2RangeID` atomic?

Reply:
This passed a few minutes of race runs so I assumed it was safe from deadlocking. It certainly makes more sense to use an atomic here, updated.